src/HeapedVectors.jl

```diff
@@ -0,0 +1,165 @@
+__precompile__()
+
+module HeapedVectors
+
+import Base: getindex, length, push!, isempty,
+        pop!, resize!, popfirst!
+
+using IntervalArithmetic
+export Strategy,heap,HeapedArray,heapVectors,SortedArray,sorVectors 
+
+abstract type Strategy end
+struct heapVectors <:Strategy end
+struct sorVectors <:Strategy end
+
+const SortedArray = sorVectors()
```

Comment:
These should not be necessary.

Reply:
But according to my logic they are needed here.

Reply:
You should just make the types `SortedVector <: Strategy` and `Heap <: Strategy`.

Reply:
Actually David I was thinking to put this `Strategy` term in `optimise.jl`  because after all it is the only place where it is using so I put `Strategy` there.